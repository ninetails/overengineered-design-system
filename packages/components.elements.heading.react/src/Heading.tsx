import React, { FunctionComponent, ReactNode } from 'react'
import styled from 'styled-components/macro'

interface HeadingTagProps {
  /** testing Storybook description */
  readonly color?: String
}

interface HeadingProps extends HeadingTagProps {
  children?: ReactNode
  description?: String
}

const HeadingTag = styled(({ color, ...props }) => <h1 {...props} />)`
  color: ${({ color }) => color};
`

HeadingTag.defaultProps = {
  color: 'red'
}

const Heading: FunctionComponent<HeadingProps> = ({
  children,
  color,
  description = 'default description'
}: HeadingProps = {}) => {
  return (
    <HeadingTag color={color} data-desc={description}>
      {children}
    </HeadingTag>
  )
}

export default Heading
