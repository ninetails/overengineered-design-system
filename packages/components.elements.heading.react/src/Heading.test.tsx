import * as React from 'react'
import { render } from '@testing-library/react'
import Heading from '.'

describe('Heading', () => {
  it('should render [snapshot]', () => {
    const { container } = render(<Heading>Children</Heading>)
    expect(container.firstChild).toMatchSnapshot()
    expect([]).toEqual(expect.toBeArray())
    expect(container.firstChild).toHaveAttribute('data-desc')
    expect(container.firstChild).not.toHaveAttribute('color')
  })
})
