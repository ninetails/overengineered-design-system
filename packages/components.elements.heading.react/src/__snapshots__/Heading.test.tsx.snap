// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render [snapshot] 1`] = `
.c0 {
  color: red;
}

<h1
  class="c0"
  data-desc="default description"
>
  Children
</h1>
`;
